mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts

```diff
@@ -15,9 +17,50 @@ export class WayPreviewDAL {
    */
   public static async getWaysPreview(): Promise<WayPreview[]> {
     const waysDTO = await WayService.getWaysDTO();
-    const waysUuids = waysDTO.map((item) => item.uuid);
 
-    const waysPreview = await Promise.all(waysUuids.map(WayPreviewDAL.getWayPreview));
+    const allNeededUsersUuids = new Set(waysDTO.flatMap(wayDTO => [
+      wayDTO.ownerUuid,
+      ...wayDTO.mentorUuids,
+      ...wayDTO.mentorRequestUuids,
+    ]));
```

Comment:
I remember i proposed to use Set, but I see below a lot om find methods, so maybe it will be better to use helper ArrayToHashMap

Reply:
Replaced with filter